src/consensus/aft/raft.h

```diff
@@ -555,6 +560,28 @@ namespace aft
       create_and_remove_node_state();
     }
 
+    void add_network_configuration(
+      ccf::SeqNo seqno, const kv::NetworkConfiguration& config)
+    {
+      LOG_DEBUG_FMT("Configurations: new network config: {{{}}}", config);
+      std::unique_lock<std::mutex> guard(state->lock, std::defer_lock);
+
+      if (is_bft_reexecution() && threading::ThreadMessaging::thread_count > 1)
+      {
+        guard.lock();
+      }
+
+      // Hooks may be reordered, so the node info in `configurations` and
```

Comment:
I don't understand this comment. Consensus hooks are always executed in TxID order. What's not currently predictable is the relative order of hooks on different maps at the same TxID. Is that what you are referring to? If so, this isn't set in stone, and could be changed.

Reply:
Yes, that's what I meant to say. I assumed that was set in stone, so I didn't look into it any further. It would be very convenient if we could make sure that the new table hook follows the nodes table hook.

Reply:
Ok, we can look at that in a separate PR.